arm/Microsoft.MachineLearningServices/workspaces/computes/deploy.bicep

```diff
@@ -0,0 +1,136 @@
+param deployCompute bool = true
+
+@sys.description('Optional. Location for the underlying compute. Ignored when attaching a compute resource, i.e. when you provide a resource ID.')
+param computeLocation string = resourceGroup().location
+
+@sys.description('Optional. The description of the Machine Learning compute.')
+param description string = ''
+
+@sys.description('Optional. Opt-out of local authentication and ensure customers can use only MSI and AAD exclusively for authentication.')
+param disableLocalAuth bool = false
+
+@sys.description('Optional. ARM resource ID of the underlying compute.')
+param resourceId string = ''
+
+@sys.description('Required. Set the object type.')
+@allowed([
+  'AKS'
+  'AmlCompute'
+  'ComputeInstance'
+  'Databricks'
+  'DataFactory'
+  'DataLakeAnalytics'
+  'HDInsight'
+  'Kubernetes'
+  'SynapseSpark'
+  'VirtualMachine'
+])
+param computeType string
+
+@sys.description('Optional. The properties of the compute. Will be ignored in case "resourceId" is set.')
+param properties object = {}
+
+@sys.description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+// Identity
+@sys.description('Optional. Enables system assigned managed identity on the resource. Ignored when attaching a compute resource, i.e. when you provide a resource ID.')
+param systemAssignedIdentity bool = false
+
+@sys.description('Optional. The ID(s) to assign to the resource. Ignored when attaching a compute resource, i.e. when you provide a resource ID.')
+param userAssignedIdentities object = {}
+
+// ================//
+// Variables       //
+// ================//
+var identityType = systemAssignedIdentity ? (!empty(userAssignedIdentities) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(userAssignedIdentities) ? 'UserAssigned' : 'None')
+
+var identity = identityType != 'None' ? {
+  type: identityType
+  userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : any(null)
+} : any(null)
+
+// ============================= //
+// Existing resources references //
+// ============================= //
+resource machineLearningWorkspace 'Microsoft.MachineLearningServices/workspaces@2021-04-01' existing = {
+  name: machineLearningWorkspaceName
+}
+
+// =========== //
+// Deployments //
+// =========== //
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource machineLearningWorkspaceCompute 'Microsoft.MachineLearningServices/workspaces/computes@2022-01-01-preview' = if (deployCompute == true) {
```

Comment:
Why the condition?

Reply:
@MrMCake As written in the README doing a deployment for an attached compute (e.g. AKS) multiple times will fail, i.e. the first one will succeed and attach the resource, but as the operation is not idempotent subsequent resources will fail. So, when you attach a compute you need to set the flag to false after the first deployment. For created resources via AML this flag is not relevant as the operation is idempotent. The PG is aware of the problem but so far has not resolved it.

Reply:
ah right, my bad